sqlx-core/src/ext/async_stream.rs

```diff
@@ -103,7 +103,7 @@ impl<'a, T> Stream for TryAsyncStream<'a, T> {
             return Poll::Ready(None);
         }
 
-        match self.future.poll_unpin(cx) {
+        match Pin::new(&mut self.future).poll(cx) {
```

Comment:
Preliminary comment: I feel like this is actually a small net-negative to code readability. If there is a real improvement in compile- or run-time performance to getting rid of `FutureExt` here, I would almost prefer if we created our own `FutureExt` trait to stick actually-useful methods on.

Reply:
I've reverted it for now. This series of PRs doesn't completely remove `futures_util` from the dependency tree, so it's not worth it to start copying their traits into sqlx if we're not going to be able to remove it completely for now anyways.